server/src/main/java/org/opensearch/index/shard/CheckpointRefreshListener.java

```diff
@@ -40,7 +40,10 @@ public void beforeRefresh() throws IOException {
 
     @Override
     public void afterRefresh(boolean didRefresh) throws IOException {
-        if (didRefresh && shard.state() == IndexShardState.STARTED && shard.getReplicationTracker().isPrimaryMode()) {
+        if (didRefresh
+            && shard.state() == IndexShardState.STARTED
+            && shard.getReplicationTracker().isPrimaryMode()
+            && !shard.indexSettings.isRemoteStoreEnabled()) {
```

Comment:
I'd prefer we not wire up the refresh listener from IndexShard in this case.

Reply:
probably add an assert for possible checks here?